pkg/services/rpcsrv/server.go

```diff
@@ -1527,7 +1526,7 @@ func (s *Server) verifyProof(ps params.Params) (any, *neorpc.Error) {
 	}
 	var p result.ProofWithKey
 	if err := p.FromString(proofStr); err != nil {
-		return nil, neorpc.ErrInvalidParams
+		return nil, neorpc.ErrInvalidProof
 	}
 	vp := new(result.VerifyProof)
 	val, ok := mpt.VerifyProof(root, p.Key, p.Proof)
```

Comment:
IMO, if `p.FromString()` fails, then it's `ErrInvalidParams`. But afterwards, if `mpt.VerifyProof` returns `false`, then it's `ErrInvalidProof`. However, this change will make us incompatible with C# node, because currently no error is returned on invalid proof (but we've agreed that it's OK for us).

@roman-khimov?

Reply:
> if p.FromString() fails, then it's ErrInvalidParams

Correct.

> if mpt.VerifyProof returns false, then it's ErrInvalidProof

Even though https://docs.neo.org/docs/en-us/reference/rpc/latest-version/api/verifyproof.html is not very helpful, it looks to me that
https://github.com/neo-project/neo-modules/blob/99ffc846e4df93491d1d812cb4c6ef6ee875e056/src/StateService/StatePlugin.cs#L255-L257 works this way already (with a different error, but with an error). It seems like `verifyproof` is just a bit outdated for some reason.

Reply:
Just to clarify.
I'll change this

p.FromString() fails, then it's ErrInvalidParams .

After that I will add an error if VerifyProof returns false, but in separate commit because it changes function behaviour. Am I right?

Reply:
Yes.